### PR TITLE
Index adress 10010

### PR DESCRIPTION
**Pull request by ivanPyrohivskyi**
Ignoring cities that never store to "city" table. Otherwise it is occur streets with NULL city in "street" table.

### CODE REVIEW

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java [hunk 1]
```diff
@@ -620,7 +624,11 @@ public int compare(City c1, City c2) {
 			}
 		});
 		for (City c : nearestObjects) {
-			if (relativeDistance(location, c) > 0.2) {
+			if(!c.getType().storedAsSeparateAdminEntity())
+				//ignore "city" thar never store in city table, see of IndexCreator.java methods: writeCitiesIntoDb && parameters of calling createMissingCity
+				continue;
+			
+			if (relativeDistance(location, c) > 0.2) { //TODO what is magic 0.2 number ???
```
Comment (vshcherb) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java:631]: No TODO in final code

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java [hunk 2]
```diff
@@ -602,6 +602,10 @@ private String putSuffixToEnd(String name, int ind, int suffixLength) {
 		nearestObjects.addAll(cityVillageManager.getClosestObjects(location.getLatitude(), location.getLongitude()));
 		//either we found a city boundary the street is in
 		for (City c : nearestObjects) {
+			if(!c.getType().storedAsSeparateAdminEntity())
```
Comment (vshcherb) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java:605]: formatting

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java [hunk 3]
```diff
@@ -602,6 +602,10 @@ private String putSuffixToEnd(String name, int ind, int suffixLength) {
...
+				//ignore "city" thar never store in city table, see of IndexCreator.java methods: writeCitiesIntoDb && parameters of calling createMissingCity
```
Comment (vshcherb) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java:606]: Comment would be:
// ignore districts, boroughs, ... so we don't get streets missing in the final obf

**Comment by ivanPyrohivskyi** [comment]
Comment fixed

**Comment by ivanPyrohivskyi** [comment]
Added { } to if statement

File: java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java [hunk 4]
```diff
@@ -625,10 +625,10 @@ public int compare(City c1, City c2) {
 		});
 		for (City c : nearestObjects) {
 			if(!c.getType().storedAsSeparateAdminEntity())
-				//ignore "city" thar never store in city table, see of IndexCreator.java methods: writeCitiesIntoDb && parameters of calling createMissingCity
+				//ignore districts, boroughs, neighbourhood, so we don't get streets missing in the final obf
```
Comment (vshcherb) [inline java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java:629]: formatting, braces

Comment (vshcherb) [reply java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexAddressCreator.java:629]: apply java formatting to block -> "if("

**Comment by ivanPyrohivskyi** [comment]
Applied Java formatting

**Pull request merged**